packages/core/src/domain/automaticErrorCollection.ts

```diff
@@ -78,8 +83,13 @@ export function stopRuntimeErrorTracking() {
 }
 
 export function trackNetworkError(configuration: Configuration, errorObservable: ErrorObservable) {
-  startXhrProxy().onRequestComplete((context) => handleCompleteRequest(RequestType.XHR, context))
-  startFetchProxy().onRequestComplete((context) => handleCompleteRequest(RequestType.FETCH, context))
+  const xhrProxy = startXhrProxy()
+  xhrProxy.beforeSend(addFocusToContext)
+  xhrProxy.onRequestComplete((context) => handleCompleteRequest(RequestType.XHR, context))
```

Comment:
I am not really fond of scattering the foreground logic everywhere.
Could we centralized the collection of foreground period and request what we need when building the event?
something like:
```
Foreground
  getPeriods(start, end)
  getStatus(time)

From view collection:
  view.foreground_periods = Foreground.getPeriods(view.start, view.end).map(toServerDuration)

From error/action collection:
  view.in_foreground = Foreground.getStatus(event.start)
```

Reply:
Feels weird, if we were to follow that logic, we would need a URL manager used in view, resources & network error.